core/deployment/src/main/java/io/quarkus/deployment/logging/LoggingResourceProcessor.java

```diff
@@ -213,17 +218,27 @@ public void accept(LogRecord logRecord, Consumer<LogRecord> logRecordConsumer) {
                 Throwable c = logRecord.getThrown();
                 while (c != null) {
                     StackTraceElement[] stackTrace = c.getStackTrace();
+                    int cutPoint = -1;
                     for (int i = 0; i < stackTrace.length; ++i) {
                         var elem = stackTrace[i];
                         if (index.getClassByName(DotName.createSimple(elem.getClassName())) != null) {
-                            stackTrace[i] = new StackTraceElement(elem.getClassLoaderName(), elem.getModuleName(),
-                                    elem.getModuleVersion(),
-                                    MessageFormat.UNDERLINE + MessageFormat.BOLD + elem.getClassName()
-                                            + MessageFormat.NO_UNDERLINE + MessageFormat.NO_BOLD,
-                                    elem.getMethodName(), elem.getFileName(), elem.getLineNumber());
+                            if (logBuildTimeConfig.highlightAppCode
+                                    && launchModeBuildItem.getLaunchMode() == LaunchMode.DEVELOPMENT) {
+                                stackTrace[i] = new StackTraceElement(elem.getClassLoaderName(), elem.getModuleName(),
+                                        elem.getModuleVersion(),
+                                        MessageFormat.UNDERLINE + MessageFormat.BOLD + elem.getClassName()
+                                                + MessageFormat.NO_UNDERLINE + MessageFormat.NO_BOLD,
+                                        elem.getMethodName(), elem.getFileName(), elem.getLineNumber());
+                            }
+                            cutPoint = i;
                         }
                     }
                     restore.put(c, c.getStackTrace());
+                    if (cutPoint != -1 && logBuildTimeConfig.trimStackTraces) {
```

Comment:
Should we leave a “…” where we cut as indicator it got stripped ?

Reply:
I don't think I can really do that, without writing a new formatter for jboss logging.

Reply:
This might have already been considered but perhaps this could be achieved by appending a new `StackTraceElement` to the result array?